zcash_client_memory/src/mem_wallet/wallet_write.rs

```diff
@@ -299,14 +308,77 @@ impl WalletWrite for MemoryWalletDb {
         &mut self,
         transactions: &[SentTransaction<Self::AccountId>],
     ) -> Result<(), Self::Error> {
-        todo!()
+        for sent_tx in transactions {
+            self.tx_table.put_tx_data(
+                sent_tx.tx(),
+                Some(sent_tx.fee_amount()),
+                Some(sent_tx.target_height()),
+            );
+            // Mark sapling notes as spent
+            if let Some(bundle) = sent_tx.tx().sapling_bundle() {
+                for spend in bundle.shielded_spends() {
+                    self.mark_sapling_note_spent(*spend.nullifier(), sent_tx.tx().txid());
+                }
+            }
+            // Mark orchard notes as spent
+            if let Some(_bundle) = sent_tx.tx().orchard_bundle() {
+                #[cfg(feature = "orchard")]
+                {
+                    for action in _bundle.actions() {
+                        self.mark_orchard_note_spent(*action.nullifier(), sent_tx.tx().txid());
+                    }
+                }
+
+                #[cfg(not(feature = "orchard"))]
+                panic!("Sent a transaction with Orchard Actions without `orchard` enabled?");
+            }
+            // Mark transparent UTXOs as spent
+            #[cfg(feature = "transparent-inputs")]
+            for utxo_outpoint in sent_tx.utxos_spent() {
+                // self.mark_transparent_utxo_spent(wdb.conn.0, tx_ref, utxo_outpoint)?;
+                todo!()
+            }
+
+            for output in sent_tx.outputs() {
+                // insert sent output
+
+                // TODO: Do we actually have to store these notes and outputs?
```

Comment:
They do need to be added here so the wallet summary will update and the wallet will appear responsive. Zcash block time is really slow so otherwise the balance won't update for minutes and trying to build new transactions in this time will respend the same outputs.

We need to ensure the scanning logic can handle seeing the same note multiple times (once from here then once when scanning)